toolchain/check/facet_type.cpp

```diff
@@ -37,16 +38,57 @@ static auto WitnessAccessMatchesInterface(
   return false;
 }
 
-auto ResolveFacetTypeImplWitness(
+static auto IncompleteFacetTypeBuilder(Context& context, SemIRLoc loc,
+                                       SemIR::InstId facet_type_inst_id,
+                                       bool is_definition)
+    -> DiagnosticBuilder {
+  // The other case is "impl as incomplete facet type with rewrites", but
+  // currently all incomplete facet types with rewrites trigger errors before
+  // this.
+  CARBON_CHECK(is_definition);
+  CARBON_DIAGNOSTIC(ImplAsIncompleteFacetTypeDefinition, Error,
+                    "impl as incomplete facet type {0} definition",
```

Comment:
i have trouble reading this grammatically, so suggesting something else:
 ```suggestion
                    "`impl as` target has facet type {0} that is not yet defined",
```

Reply:
I was trying to convey that `impl` definitions specifically require the facet type to be complete (now unlike `impl` declarations).

Reply:
Maybe something like `definition of impl as incomplete facet type {0}`?